snuba/cli/multistorage_consumer.py

```diff
@@ -259,6 +262,44 @@ def stats_callback(stats_json: str) -> None:
 
         commit_log_config = CommitLogConfig(producer, commit_log, consumer_group)
 
+    strategy_factory = build_multistorage_streaming_strategy_factory(
+        storages,
+        max_batch_size,
+        max_batch_time_ms,
+        processes,
+        input_block_size,
+        output_block_size,
+        metrics,
+        commit_log_config,
+    )
+
+    configure_metrics(StreamMetricsAdapter(metrics))
+    processor = StreamProcessor(
+        consumer,
+        topic,
+        strategy_factory,
+        IMMEDIATE,
+    )
+
+    def handler(signum: int, frame: Any) -> None:
+        processor.signal_shutdown()
+
+    signal.signal(signal.SIGINT, handler)
+    signal.signal(signal.SIGTERM, handler)
+    processor.run()
+
+
+def build_multistorage_streaming_strategy_factory(
+    storages: Dict[Any, WritableTableStorage],
```

Comment:
This type could probably be more specific than Any. Though would it make sense to pass a sequence of storage keys here instead of a dict? Since we already get a sequence of those from both the consumer and multistorage CLI, it could make sense to pass it straight into this function.

Reply:
Hm, so I changed the type of the arg that gets passed into the build function. Were you suggesting to remove the `storages` dictionary altogether? 

I agree about validating the topics beforehand so we don't build the factory unless necessary. I think that may need to live in a different function though as it might be a fairly long function with different topic validations and/or resolutions.

Reply:
It could still be pulled into a separate function. As long as that separate function is called from within `build_multistorage_streaming_strategy_factory` and not outside of it.

Reply:
Yep, that makes sense, that would be my next PR.